scripts/database/clean-datasets-node/DeleteOldData.js

```diff
@@ -0,0 +1,96 @@
+var pg = require('pg');
+var moment = require('moment');
+//config
+var pgclient = new pg.Client({
+  host: 'localhost',
+  database: 'hoot',
+  port: 5432,
+  user: 'hoot',
+  password: 'hoottest'
+});
+
+// disconnect
+function disconnect() {
+pgclient.end(function(err){
+  console.log('disconnected');
+  if (err) {
+    console.log('error during disconnection', err.stack)
+  }
+});
+};
+
+pgclient.connect(function(err){
+  if (err) {
+    console.error('connection error', err.stack);
+  } else {
+    // client has connected to pg
+    console.log('connected');
+    // get dataset ids, names, and lastAccessed dates from pg table
+    var getStaleData = pgclient.query("select id, display_name, (a).value from ( \
+                                  select *, each(tags) as a \
+                                  from maps \
+                                ) t;");
+    var rows = [];
+    getStaleData.on('row', function(row, result) {
+      rows.push(row);
+    });
+    getStaleData.on('end', function(result){
+      if(rows[0] === undefined){
+        // if no results are returned, disconnect
+        console.log('no data returned');
+      } else {
+        // create new array with data id , last access date, and whether or not it's stale
+        var data = rows.map(function(row){
+          var stale = moment().diff(moment(row.value), 'days') > 60;
+          return {
+            id: row.id,
+            name: row.display_name,
+            accessed: row.value,
+            stale: stale
+          }
+        });
+        // delete stale datasets
+        data.forEach(function(d){
+          if (d.stale) {
+            pgclient.query("delete from maps where id=" + d.id);
+            // not sure if these are necessary
+            // pgclient.query("delete from folder_map_mappings where map_id=" + d.id);
+            // pgclient.query("drop table current_nodes_" + d.id + " cascade");
+            // pgclient.query("drop sequence changesets_" + d.id + "_id_seq");
+            // pgclient.query("drop table current_nodes_" + d.id + "cascade");
+            // pgclient.query("drop sequence current_nodes_" + d.id + "_id_seq");
+            // pgclient.query("drop table current_relation_members_" + d.id + "cascade");
+            // pgclient.query("drop sequence current_relations_" + d.id + "_id_seq");
+            // pgclient.query("drop table current_relations_" + d.id + "cascade");
+            // pgclient.query("drop table current_way_nodes_" + d.id + "cascade");
+            // pgclient.query("drop table current_ways_" + d.id + "cascade");
+            // pgclient.query("drop sequence current_ways_" + d.id + "_id_seq");
+            console.log("deleted dataset " + d.name);
```

Comment:
Should also drop "renderdb_mapid" db that goes with dataset.  Not sure if the commented out tables constitute the entirety of a hoot map dataset.  Could always shell out to a `hoot delete-map` command.

Reply:
That would be easy enough to add :) did you try the script tho?

Reply:
Not yet :bee:

Reply:
:eyes: